ado_wrapper/resources/runs.py

```diff
@@ -97,22 +111,22 @@ def run_all_and_capture_results_sequentially(
         Once done, returns a mapping of definition_id -> `Run` object"""
         return_values = {}
         for definition_id, build_def_data in data.items():
-            template_variables, branch_name = build_def_data["template_variables"], build_def_data["branch_name"]
-            run = cls.run_and_wait_until_completion(ado_client, definition_id, template_variables, branch_name, max_timeout_seconds)
+            template_variables, branch_name, stages_to_run = build_def_data["template_variables"], build_def_data["branch_name"], build_def_data["stages_to_run"]
+            run = cls.run_and_wait_until_completion(ado_client, definition_id, template_variables, branch_name, max_timeout_seconds, stages_to_run)
             return_values[definition_id] = run
         return return_values
 
     @classmethod
     def run_all_and_capture_results_simultaneously(
-        cls, ado_client: "AdoClient", data: dict[str, RunAllDictionary], max_timeout_seconds: int | None = 1800
+        cls, ado_client: "AdoClient", data: dict[str, RunAllDictionary], max_timeout_seconds: int | None = 1800, stages_to_run: list[str] | None = None
     ) -> dict[str, "Run"]:
-        """Takes a mapping of definition_id -> {template_variables, branch_name}
-        Once done, returns a mapping of definition_id -> `Run` object"""
+        """Takes a mapping of stage_internal_name -> RunAllDictionary
```

Comment:
It doesn't take the stage internal name, it takes a mapping of definition id to RunAllDictionary, and returns a mapping of def_id